modules/apps/transfer/keeper/relay_test.go

```diff
@@ -261,17 +406,102 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestOnRecvPacketDoesNotSetTotalEscrowAmountForSourceIBCToken() {
+	/*
+		Given the following flow of tokens:
+
+		chain A (channel 0) -> (channel-0) chain B (channel-1) -> (channel-1) chain A (channel-1)             -> (channel-1) chain B
```

Comment:
Why can't we just do chainA -> chanB and check onRecv on chainB instead of doing chainA->ChainB->ChainA->ChainB to test this feature?

Reply:
Because we should test the scenario of receiving non-native IBC tokens in the source chain. If you only do chainA -> chanB, the vouchers received on B are non-native IBC, but the chain is not the source. Do you agree?

Reply:
yup i agree this makes sense